### PR TITLE
Pools in Elastic (Rough Cut)

### PR DESCRIPTION
**Pull request by vantuz-subhuman**
Rought cut of  Shelley chain exporting into ES.
1. Updated how entities are constructed for ES to allow for Shelley stuff
2. Fixed many small things in how we are processing ES operations in general
3. Expanded Shelley cert parsers with new available fields
4. Dumping pools additionally into txs and also tracking delegation sum per account and per pool
5. Pool owners meta from GitHub is dumped into another index

⚠️ This branch does not implement full Shelley support in Elastic yet, neither it does all the things properly. But it is decided to merge it, because it contains many fixes and changes already, so it will be easier to build full support from this foundation.

### CODE REVIEW

File: src/blockchain/shelley/utils.js [hunk 1]
```diff
@@ -4,25 +4,66 @@ import type { PoolRegistrationType, PoolRetirementType, StakeDelegationType } fr
...
 
 // frees input rust-wasm Value and parses it into a js number
-const consumeValueToNumber = (value: any): number => {
+const consumeOptionalValueToNumber = (value: any): number => {
   // TODO: Values are returned as strings under the rationale that js strings
   // can only fit a 52-bit radix as integers, but since the max ADA supply is smaller
   // than this (but bigger than a 32-bit int) this should be safe. We should try and
   // see if this can be changed in js-chain-libs and use that there instead.
+  if (!value) {
+    return null
```
Comment (rooooooooob) [inline src/blockchain/shelley/utils.js]: Is the plan to just pretend that it's `number` and not `?number` as the return type? If not, then we'd be wanting to add in explicit checks on the return. Where were we getting null inputs anyway?

File: src/blockchain/shelley/utils.js [hunk 2]
```diff
@@ -139,21 +180,32 @@ const fragmentToObj = (fragment: any, networkDiscrimination: number, extraData:
           type: CERT_TYPE.PoolRegistration,
           pool_id: poolId.to_string(),
           // we should be able to do this considering js max int would be 285,616,414 years
-          start_validity: parseInt(startValidity.to_string(), 10),
-          owners: pool_owners,
+          start_validity: parseInt(reg.start_validity().to_string(), 10),
```
Comment (rooooooooob) [inline src/blockchain/shelley/utils.js]: I think that this was a merge error? `start_validity` returns a wasm bound struct, causing a leak here.

File: src/blockchain/shelley/utils.js [hunk 3]
```diff
@@ -4,25 +4,66 @@ import type { PoolRegistrationType, PoolRetirementType, StakeDelegationType } fr
...
 
 // frees any generic rust-wasm id (anything with as_bytes()) and creates a hex string buffer from it
 const consumeIdToHex = (id: any): string => {
   const hex = Buffer.from(id.as_bytes()).toString('hex')
-  id.free()
+  free(id)
   return hex
 }
 
+// frees any generic rust-wasm id (anything with as_bytes()) and creates a hex string buffer from it
```
[the remark this answers is not in the document]
Comment (vantuz-subhuman) [reply src/blockchain/shelley/utils.js]: 😬 👍

File: src/blockchain/shelley/block.js [hunk 4]
```diff
@@ -57,6 +57,10 @@ export default class ShelleyBlock implements Block {
     return this.prevHash
   }
 
+  isGenesisBlock() {
+    return this.prevHash === '0000000000000000000000000000000000000000000000000000000000000000'
```
[the remark this answers is not in the document]
Comment (vantuz-subhuman) [reply src/blockchain/shelley/block.js]: 🤔 Good point!

File: src/blockchain/shelley/utils.js (hunk 1 above)
Comment (vantuz-subhuman) [reply src/blockchain/shelley/utils.js]: Yep, incorrect return type. Value might be null in the pool-registration cert in the `limit` parameter, when `max_limit()` is queried from Rust object. That field is optional and might null.

I'm changing the return type to `?number` 👌

File: src/blockchain/shelley/utils.js (hunk 2 above)
Comment (vantuz-subhuman) [reply src/blockchain/shelley/utils.js]: Yep, fixed to call the function

**Pull request merged**